test/controllers/DeleteSchemeControllerSpec.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import connectors.DataCacheConnector
+import controllers.actions._
+import forms.DeleteSchemeFormProvider
+import org.mockito.Matchers.any
+import org.mockito.Mockito.when
+import org.scalatest.mockito.MockitoSugar
+import play.api.data.Form
+import play.api.mvc.Results.Ok
+import play.api.test.Helpers.{contentAsString, _}
+import utils.FakeNavigator
+import views.html.deleteScheme
+
+import scala.concurrent.Future
+
+class DeleteSchemeControllerSpec extends ControllerSpecBase with MockitoSugar {
+
+  val formProvider = new DeleteSchemeFormProvider()
+  val form: Form[Boolean] = formProvider()
+  val schemeName = "Test Scheme Name"
+  val fakeDataCacheConnector = mock[DataCacheConnector]
+
+  def controller(dataRetrievalAction: DataRetrievalAction = getMandatorySchemeName): DeleteSchemeController =
+    new DeleteSchemeController(frontendAppConfig, messagesApi, fakeDataCacheConnector, FakeNavigator, FakeAuthAction,
+      dataRetrievalAction, new DataRequiredActionImpl, formProvider)
+
+  def viewAsString(form: Form[_] = form): String = deleteScheme(frontendAppConfig, form, schemeName)(fakeRequest, messages).toString
+
+  "DeleteScheme Controller" must {
+
+    "return OK and the correct view for a GET" in {
+      when(fakeDataCacheConnector.removeAll(any())(any(), any())).thenReturn(Future.successful(Ok))
```

Comment:
In the tests for DeleteSchemeController in testing the Get/OnPageLoad there is a When(fakeController,removeAll ... ) but the OnPageLoad doesn't call the RemoveAll so I'm wondering if it was intended for tests for the OnSubmit method which does invoke this method, but maybe the OnSubmit test needs an additional assertion to show that the RemoveAll was called?